versioned/spi/src/main/java/org/projectnessie/versioned/CommitResult.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface CommitResult<COMMIT> extends Result {
+
+  @Override
+  default ResultType getResultType() {
+    return ResultType.COMMIT;
+  }
+
+  COMMIT getCommit();
```

Comment:
Nit: add a `getCommitHash` convenience method (to avoid all the added `.getCommit().getHash()` chains)

Reply:
It's a bit tricky because `COMMIT` can be anything, but I'll find a way.